src/base/static/components/form-fields/form-field.js

```diff
@@ -108,4 +121,9 @@ FormField.propTypes = {
   ]),
 };
 
+FormField.defaultProps = {
+  attachmentModels: [],
+  modelId: -1,
```

Comment:
We may or may not have a `modelId` passed to a `FormField` component (we'll have one if the `FormField` is instantiated as part of the detail view editor--where a model already exists--but we won't if it's instantiated as part of the input form). We use the `modelId` prop when validating custom URLs, and I was thinking this default prop of `-1` was a concise way to communicate that no existing model is present. Do you think there's a clearer way to express this?

You're right that the `attachmentModels` default prop isn't necessary.

Reply:
Sounds good - that seems like a valid use case for the default props.

But I wonder - does `modelId` PropTypes need to be required then? Also, can we just rely on it being `undefined` instead of setting it to `-1`?

Reply:
Ah, that's true-- it could just fall back to being `undefined`.